### PR TITLE
HTTPS devServer option

### PR DESCRIPTION
**Pull request by Fjandin**
Added option to start devServer in HTTPS

`https?: https.ServerOptions`

example:
```js
fuse.dev({
        root: 'dist',
        hmr: true,
        https: {
            key: fs.readFileSync('./cert/app.key', 'utf8'),
            cert: fs.readFileSync('./cert/app.crt', 'utf8')
        }
    })
```

### CODE REVIEW

**Comment by Fjandin** [comment]
Alrighty. Added.

Took the liberty of adding a missing documentation of the `root` option of `CSSModules`.

If you want this in a seperate PR I can do this

**Comment by Kompwu** [comment]
If you can separate that. that would be perfect.

**Comment by Fjandin** [comment]
I have removed the changes to CSSModules docs. Will open a seperate PR for this

File: src/devServer/HTTPServer.ts [hunk 1]
```diff
@@ -74,6 +77,16 @@ Development server running http://localhost:${port} @ ${packageInfo.version}
         return socketServer;
     }
 
+    private createServer(): http.Server | https.Server {
```
Comment (Kompwu) [inline src/devServer/HTTPServer.ts]: I almost forgot...
Can you write tests for that?

Comment (Fjandin) [reply src/devServer/HTTPServer.ts]: Wrote a test for the https option... tbh i'm a bit lost in the test directory. I don't seem to find anywhere where the HTTPserver is tested

Comment (Kompwu) [reply src/devServer/HTTPServer.ts]: I don't think there was any.
It's looks fine for me.
Let's wait @nchanged advice.

**Pull request merged**